### ISSUE TITLE
Query not working

### ISSUE THREAD

**Issue by kevinxin90** [biothings/biothings_explorer#24]
```
{
  "message": {
    "query_graph": {
      "nodes": {
        "n00": {
          "id": "MONDO:0002715",
          "category": "biolink:Disease"
        },
        "n01": {
          "category": "biolink:ChemicalSubstance"
        },
        "n02": {
          "category": "biolink:Gene"
        }
      },
      "edges": {
        "e00": {
          "predicate": "biolink:correlated_with",
          "subject": "n00",
          "object": "n01"
        },
        "e01": {
          "predicate": "biolink:related_to",
          "subject": "n01",
          "object": "n02"
        }
      }
    }
  }
}
```

Error:
```
{
    "error": "TypeError: Cannot convert undefined or null to object"
}
```

**Comment by kevinxin90** [comment]
The issue is caused because:
1. The first part of the query: Disease -> correlated_with -> ChemicalSubstance returns empty results.
2. However, bte doesn't terminate after the first query fails.

Solution:

BTE should terminate whenever one intermediate query step fails.

**Comment by kevinxin90** [comment]
Fixed in: b1b56227f79464658e6f0ea9571edc25126472b5
